api/mesh/options.proto

```diff
@@ -40,6 +40,13 @@ message KumaResourceOptions {
 
   // WsAdmin whether this entity requires admin auth to access these endpoints
   bool ws_admin = 11;
+
+  // HasCustomKumaCtlRender whether we should use a custom function to render
+  // entities in kumactl
+  bool has_custom_kumactl_render = 12;
```

Comment:
Perhaps this is better addressed by registering renderers by type and looking up the one you need in kumactl?